src/basho_bench_worker.erl

```diff
@@ -334,3 +349,27 @@ rate_worker_run_loop(State, Lambda) ->
         ExitReason ->
             exit(ExitReason)
     end.
+
+%% Not passing state API - expect non-list spec or error during new attempt
+init_generators(false, Config, Id, Module) ->
+    Module:new(Config, Id);
+%% Passing state API - process list or turn single-spec into [{default,Generator}]
+init_generators(true, ConfigList, Id, Module) -> 
+    config_to_generators(ConfigList, Id, Module, []).
+
+config_to_generators([], _Id, _Module, ACC) ->
+    ACC;
+config_to_generators([{Name, Spec}|ConfigListTail], Id, Module, ACC) ->
+    config_to_generators(ConfigListTail, Id, Module, [ {Name, Module:new(Spec, Id)} | ACC]);
+config_to_generators(Spec, Id, Module, _ACC) when is_tuple(Spec) ->
+    [{default, Module:new(Spec, Id)}].
```

Comment:
Ok so if you enable the api_state_passing option you must cut over to named generator lists for both key and value generator at the same time. That's fine.